src/Std/Data/DTreeMap/Lemmas.lean

```diff
@@ -2022,4 +2022,427 @@ theorem getD_unitOfList {l : List α} {k : α} {fallback : Unit} :
 
 end Const
 
+section Alter
+
+theorem isEmpty_alter_eq_isEmpty_erase [TransCmp cmp] [LawfulEqCmp cmp] {k : α}
+    {f : Option (β k) → Option (β k)} :
+    (t.alter k f).isEmpty = ((t.erase k).isEmpty && (f (t.get? k)).isNone) :=
+  Impl.isEmpty_alter_eq_isEmpty_erase t.wf
+
+@[simp]
+theorem isEmpty_alter [TransCmp cmp] [LawfulEqCmp cmp] {k : α}
+    {f : Option (β k) → Option (β k)} :
+    haveI : BEq Nat := instBEqOfDecidableEq
```

Comment:
It this really necessary? If so, we need to find out why this is necessary and fix it.

Reply:
Oh, I only wanted (and needed) to have this explicit instance in `Internal.Lemmas`, where it is safe (albeit ugly) w.r.t. the public API.

However, by lowering `beqOfOrd`'s priority, I was able to eliminate all of these `haveI`'s, even those used internally.